### ISSUE TITLE
[CI] MixedClusterClientYamlTestSuiteIT test {p0=synonyms/80_synonyms_from_index/Fail loading synonyms from index if synonyms_set doesn't exist} failing

### ISSUE THREAD

**Issue by arteam** [elastic/elasticsearch#99769]
**Build scan:**
https://gradle-enterprise.elastic.co/s/lsgnolasjvlly/tests/:qa:mixed-cluster:v8.9.2%23mixedClusterTest/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT/test%20%7Bp0=synonyms%2F80_synonyms_from_index%2FFail%20loading%20synonyms%20from%20index%20if%20synonyms_set%20doesn't%20exist%7D

**Reproduction line:**
```
./gradlew ':qa:mixed-cluster:v8.9.2#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=synonyms/80_synonyms_from_index/Fail loading synonyms from index if synonyms_set doesn't exist}" -Dtests.seed=211553EE91921FA8 -Dtests.bwc=true -Dtests.locale=sk-SK -Dtests.timezone=Canada/Newfoundland -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT&tests.test=test%20%7Bp0%3Dsynonyms/80_synonyms_from_index/Fail%20loading%20synonyms%20from%20index%20if%20synonyms_set%20doesn't%20exist%7D

**Failure excerpt:**
```
org.junit.AssumptionViolatedException: [synonyms/80_synonyms_from_index] skipped, reason: [Loading synonyms from index is introduced in 8.10.0]

  at com.carrotsearch.randomizedtesting.RandomizedTest.assumeTrue(RandomizedTest.java:745)
  at com.carrotsearch.randomizedtesting.RandomizedTest.assumeFalse(RandomizedTest.java:753)
  at org.apache.lucene.tests.util.LuceneTestCase.assumeFalse(LuceneTestCase.java:906)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:456)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

**Comment by cbuescher** [comment]
This probably only needs "skip" version adjustment in the test.

**Comment by carlosdelest** [comment]
I can't reproduce this locally 😢 

I see a single failure on the [last 7 days](https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Madrid&tests.container=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT&tests.test=test%20%7Bp0%3Dsynonyms/80_synonyms_from_index/Fail%20loading%20synonyms%20from%20index%20if%20synonyms_set%20doesn%27t%20exist%7D):

![image](https://github.com/elastic/elasticsearch/assets/6339205/f4c7f539-59a6-4416-b4f4-84377b058e88)

The `skip` version is correct, synonyms were included in 8.10. I'm not sure what the test is complaining about on this one.

**Comment by carlosdelest** [comment]
I'm closing this since I don't see further test failures and can't reproduce locally
